R/build-home-index.R

```diff
@@ -120,3 +121,27 @@ repo_link <- memoise(function(pkg) {
 
   NULL
 })
+
+is_url <- function(x) {
+  # https://www.regextester.com/94502
+  pat <- "^(?:http(s)?:\\/\\/)?[\\w.-]+(?:\\.[\\w\\.-]+)+[\\w\\-\\._~:/?#[\\]@!\\$&'\\(\\)\\*\\+,;=.]+$"
+  grepl(pat, x, perl = TRUE)
+}
+
+is_email <- function(x) {
+  # https://www.regextester.com/19
+  pat <- "^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$"
+  grepl(pat, x, perl = TRUE)
+}
+
+bug_report_link <- function(pkg) {
+  x <- pkg$desc$get("BugReports")[[1]]
+
+  # check that value listed in BugReports field
+  # is a url or email
+  if (is_url(x) || is_email(x)) {
+    return(x)
+  }
+  # default is maintainer email
+  pkg$desc$get_authors()[1]$email
```

Comment:
The maintainers e-mail address is
```suggestion
  pkg$desc$get_author(role = "cre")$email
```